L1Trigger/L1THGCal/interface/backend/HGCalMulticlusteringHistoClusteringImpl.h

```diff
@@ -0,0 +1,73 @@
+#ifndef __L1Trigger_L1THGCal_HGCalMulticlusteringHistoClusteringImpl_h__
+#define __L1Trigger_L1THGCal_HGCalMulticlusteringHistoClusteringImpl_h__
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "DataFormats/L1THGCal/interface/HGCalCluster.h"
+#include "DataFormats/L1THGCal/interface/HGCalMulticluster.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "L1Trigger/L1THGCal/interface/HGCalTriggerGeometryBase.h"
+#include "L1Trigger/L1THGCal/interface/backend/HGCalShowerShape.h"
+#include "L1Trigger/L1THGCal/interface/HGCalTriggerTools.h"
+#include "L1Trigger/L1THGCal/interface/backend/HGCalTriggerClusterIdentificationBase.h"
+
+
+class HGCalMulticlusteringHistoClusteringImpl{
+
+public:
+
+    HGCalMulticlusteringHistoClusteringImpl( const edm::ParameterSet &conf);    
+
+    void eventSetup(const edm::EventSetup& es) 
+    {
+        triggerTools_.eventSetup(es);
+        shape_.eventSetup(es);
+        if ( (!dr_byLayer_coefficientA_.empty() && (dr_byLayer_coefficientA_.size()-1) != triggerTools_.lastLayerBH())
+             || (!dr_byLayer_coefficientB_.empty() && (dr_byLayer_coefficientB_.size()-1) != triggerTools_.lastLayerBH())
+             ) {
+            throw cms::Exception("Configuration") << 
+                "The per-layer dR values go up to " << (dr_byLayer_coefficientA_.size()-1) << 
+              "(A) and " << (dr_byLayer_coefficientB_.size()-1) << "(B), while layers go up to " << triggerTools_.lastLayerBH() << "\n";
+        }
+    }
+
+    float dR( const l1t::HGCalCluster & clu,
+              const GlobalPoint & seed ) const;
+
+    void clusterizeHisto( const std::vector<edm::Ptr<l1t::HGCalCluster>> & clustersPtr,
+                          l1t::HGCalMulticlusterBxCollection & multiclusters,
+                          std::vector<std::pair<GlobalPoint, double> > & seedPositionsEnergy,
```

Comment:
Since it is an input argument, it should be const. And also before the output in list of arguments.